src/overlay/SurveyManager.cpp

```diff
@@ -89,10 +302,22 @@ SurveyManager::addNodeToRunningSurveyBacklog(
 
     addPeerToBacklog(nodeToSurvey);
     updateSurveyExpiration(surveyDuration);
+
+    if (type == TIME_SLICED_SURVEY_TOPOLOGY)
+    {
+        if (!inboundPeersIndex.has_value() || !outboundPeersIndex.has_value())
+        {
+            throw std::runtime_error(
+                "addNodeToRunningSurveyBacklog failed: missing peer indices");
+        }
+
+        mInboundPeerIndices[nodeToSurvey] = inboundPeersIndex.value();
```

Comment:
I think downstream scripts might misuse this API, and submit two requests for the same node but with different indexes, causing one to override the other, so we should probably return an error of sorts (before adding both requests to backlog)

Reply:
(Should we also add a test for this?)

Reply:
Good point. I added a check and a test in d75739245a69b6c728625f4662bd502fd469d829